UsageStatsPlugin.inc.php

```diff
@@ -192,10 +192,11 @@ function getInstallSitePluginSettingsFile() {
 	}
 
 	/**
-	 * @copydoc Plugin::getInstallSchemaFile()
+	 * @copydoc Plugin::getInstallMigration()
 	 */
-	function getInstallSchemaFile() {
-		return $this->getPluginPath() . DIRECTORY_SEPARATOR . 'schema.xml';
+	function getInstallMigration() {
+		$this->import('UsageStatsMigration');
+		return new UsageStatsMigration();
```

Comment:
So when a plugin is installed it will run the `getInstallMigration()` method and the table will be created, yeah?

Reply:
Yes, and this function deprecates the old `getInstallSchemaFile`, which will still be supported for a while but will eventually be removed.

Actually, if you have an opinion, I'm on the fence about this new function -- I'm tempted to require that plugins include an `install.xml` and `upgrade.xml` to match the OJS descriptors, and specify any migrations that way. It'll mean an extra file for any plugin that wants to install a schema, but it would permit multiple migrations for complex plugins (rather than just the one supported by this function) and would channel all migrations into a single mechanism. Do you have a preference?

For this plugin, `install.xml` would look something like:
```
<install version="3.2.1.0">
    <migration class="plugins.generic.usageStats.UsageStatsMigration" />
</install>
```

Reply:
I like the idea of having `install.xml` and `upgrade.xml` files. :+1:

Reply:
Thanks, @NateWr, I'll remove this function and see if plugin-based descriptors work as intended!

Reply:
I put some time into this and unfortunately got into a tangle of rat's-nesty code in charge of handling plugin installs, app installs, and upgrades, and parsing the various XML descriptors for each. I'd suggest leaving this as is for the moment, and coming back to it when we're ready to deprecate and remove the XML descriptors in favour of something more standard.